cobbler/actions/reposync.py

```diff
@@ -378,7 +378,10 @@ def rhn_sync(self, repo):
         if has_rpm_list:
             self.logger.warning("warning: --rpm-list is not supported for RHN content")
         rest = repo.mirror[6:]      # everything after rhn://
-        cmd = "%s %s --repo=%s --download_path=%s" % (cmd, self.rflags, pipes.quote(rest), pipes.quote(self.settings.webdir + "/repo_mirror"))
+        
+        # el7:         -p or --download_path
+        # el8, fedora: -p or --download-path
```

Comment:
I guess here (2nd line of the comment) --download-path is not allowed, only -p?

Reply:
The option -p works in both yum and dnf, so there will be no need to check these 2 cases in the code.

Reply:
Hm, but the comment as it is does not help much?
You would have to add where it only works with -p or am I missing something?

Reply:
Yes, in the comment, you may not immediately notice the difference between "_" and "-" in --download<x>path. But this leads to a runtime error in one of the cases. -p works in the el7, el8 and fedora.

Reply:
> Beside the one comment which looks like a copy and paste issue to me?..

dnf reposync (as in dnf in fedora or el8) uses --download-path

running: /usr/bin/dnf reposync --newest-only --delete --config=/var/www/cobbler/repo_mirror/OEL-7-Latest/.origin/OEL-7-Latest.repo --repoid=OEL-7-Latest --download_path=/var/www/cobbler/repo_mirror -a x86_64
received on stdout: 
received on stderr: usage: dnf reposync [-c [config file]] [-q] [-v] [--version]
                    [--installroot [path]] [--nodocs] [--noplugins]
                    [--enableplugin [plugin]] [--disableplugin [plugin]]
                    [--releasever RELEASEVER] [--setopt SETOPTS]
--nobest] [-C]
                    [-R [minutes]] [-d [debug level]] [--debugsolver]
                    [--showduplicates] [-e ERRORLEVEL] [--obsoletes]
                    [--rpmverbosity [debug level name]] [-y] [--assumeno]
--repo
--disable] [-x [package]]
                    [--disableexcludes [repo]] [--repofrompath [repo,path]]
                    [--noautoremove] [--nogpgcheck] [--color COLOR]
                    [--refresh] [-4] [-6] [--destdir DESTDIR] [--downloadonly]
                    [--comment COMMENT] [--bugfix] [--enhancement]
                    [--newpackage] [--security] [--advisory ADVISORY]
                    [--bz BUGZILLA] [--cve CVES]
                    [--sec-severity {Critical,Important,Moderate,Low}]
                    [--forcearch ARCH] [-a [arch]] [--delete] [-m]
                    [--download-metadata] [-n] [-p DOWNLOAD_PATH]
                    [--norepopath] [--metadata-path METADATA_PATH] [--source]
                    [--remote-time] [-u]
dnf reposync: error: unrecognized arguments: --download_path=/var/www/cobbler/repo_mirror